convene-web/spec/requests/spaces/invitations_request_spec.rb

```diff
@@ -1,12 +1,35 @@
 require 'rails_helper'
 
 RSpec.describe "/spaces/:space_id/invitations", type: :request do
-  it "doesn't expose" do
+  it "created and sends an invitation for a space" do
     client = Client.create!
     space = Space.create!(:client => client, :name => "bar")
+    mail = double(deliver_now: true)
+    allow(SpaceInvitationMailer).to receive(:space_invitation_email).and_return(mail)
 
     post "/spaces/#{space.slug}/invitations", :params => { :invitation => {:name => "foobar", :email => "foobar@example.com"} }
-    expect(response).to redirect_to([space, space.invitations.find_by(name: 'foobar', email: 'foobar@example.com')])
-    expect(space.invitations.find_by(name: 'foobar', email: 'foobar@example.com')).to be_present
+
+    invitation = space.invitations.find_by(name: 'foobar', email: 'foobar@example.com')
+
+    expect(response).to redirect_to([space, invitation])
+    expect(invitation).to be_present
+
+    # TODO finish building logic to get below assertions to pass
+    # generate mailer and views
+    expect(SpaceInvitationMailer).to have_received(:space_invitation_email).with(invitation)
+    expect(mail).to have_received(:deliver_now)
+    expect(invitation.status).to eq("sent")
   end
+
+  # ------ #
+  # NOTES
+  # ------ #
+  # Kelly H, Zee, Egbet decided these are the possible invitation statuses:
+  # pending - email was not sent yet due to mailer issues etc.
+  # sent - email was sent but not accepted
+  # accepted - receiver accepted the invitation
+  # rejected - receivers rejected the invitation
+
```

Comment:
Have been thinking about adding an additional invitation status of `resent` that logs the amount of times resent, default being `0`. I hear Zee in the back of my head saying this may be getting ahead of myself, but with a query like `space.invitations.find_by(name: 'foobar', email: 'foobar@example.com')` we can run into an issue where an invite is sent to the same name/email more than once and this query not working. I think we may have actually seen this happen in our last session where there were two invitation records of the same name/email and we saw an error relating to expecting to find just one record.

Reply:
This is an interesting corner case to think through!

There are a number of paths we could go:

1. Ignore it for now (depending on what we anticipate the impact is)
2. Use a counter
3. Use an Event pattern to track when we send invitations; so that we don't lose context and can make more informed decisions (either programmatically or as a human)
4. ???

Reply:
I like 1. + adding it as a todo or notes somewhere for now. and 3 when we tackle eventually it.